node/bft/src/helpers/telemetry.rs

```diff
@@ -0,0 +1,368 @@
+// Copyright 2024-2025 Aleo Network Foundation
+// This file is part of the snarkOS library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use snarkvm::{
+    ledger::{
+        committee::Committee,
+        narwhal::{BatchCertificate, BatchHeader, Subdag},
+    },
+    prelude::{Address, Field, Network, cfg_iter},
+};
+
+use indexmap::{IndexMap, IndexSet};
+use parking_lot::RwLock;
+use rayon::prelude::*;
+use std::sync::Arc;
+
+// TODO: Consider other metrics to track:
+//  - Response time
+//  - Sync rate
+//  - Latest height of each validator
+//  - Percentage of proposals that are converted into certificates
+//  - Fullness of proposals
+//  - Connectivity (how many other validators are they connected to)
+//  - Various stake weight considerations
+//  - The latest seen IP address of each validator (useful for debugging purposes)
+
+/// Tracker for the participation metrics of validators.
+#[derive(Clone, Debug)]
+pub struct Telemetry<N: Network> {
```

Comment:
nit: with all the members `Arc`ed, if this object is expected to be cloned often, that operation would be faster if it was wrapped in a single `Arc` (and not to derive `Clone`); that way every clone would only involve a single atomic op instead of (currently) 4

Reply:
This object should not be cloned too often, as it should perpetually exist within gateway rather than passed around.

I think unifying Arcs can be in a subsequent PR if we decide to add additional metrics